src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -135,25 +136,25 @@ internal Guid GetTelemetryId(FixAllScope? fixAllScope = null)
         /// The sequence of operations that define the code action.
         /// </summary>
         public Task<ImmutableArray<CodeActionOperation>> GetOperationsAsync(CancellationToken cancellationToken)
-            => GetOperationsAsync(new ProgressTracker(), cancellationToken);
+            => GetOperationsAsync(originalSolution: null!, new ProgressTracker(), cancellationToken);
```

Comment:
I'm also concerned that even if the banned API more or less means we can't hit this path in our product, we still have a vulnerability for other hosts I think. Should we at least have a few tests testing the old path to ensure we don't break it if we're not using NRT to ensure safety?

Reply:
> given the methods being called are internal, can we at least annotate the internal methods correctly?

Because i don't want internal calls to ever pass null.  If we annotate as 'null' that can happen.  Annotating as non-null means we catch our own mistakes.

Reply:
One option is `[DisallowNull] Solution? originalSolution`, but I'm not sure it's better.